src/workerd/api/actor-state.c++

```diff
@@ -120,77 +120,75 @@ ActorObserver& currentActorMetrics() {
 jsg::Value listResultsToMap(jsg::Lock& js,
                             ActorCacheOps::GetResultList value,
                             bool completelyCached) {
-  v8::HandleScope scope(js.v8Isolate);
-  auto context = js.v8Context();
-
-  auto map = v8::Map::New(js.v8Isolate);
-  size_t cachedReadBytes = 0;
-  size_t uncachedReadBytes = 0;
-  for (auto entry: value) {
-    auto& bytesRef = entry.status == ActorCacheOps::CacheStatus::CACHED
-                   ? cachedReadBytes : uncachedReadBytes;
-    bytesRef += entry.key.size() + entry.value.size();
-    jsg::check(map->Set(context, jsg::v8Str(js.v8Isolate, entry.key),
-        deserializeV8Value(js, entry.key, entry.value)));
-  }
-  auto& actorMetrics = currentActorMetrics();
-  if (cachedReadBytes || uncachedReadBytes) {
-    size_t totalReadBytes = cachedReadBytes + uncachedReadBytes;
-    uint32_t totalUnits = billingUnits(totalReadBytes);
-
-    // If we went to disk, we want to ensure we bill at least 1 uncached unit.
-    // Otherwise, we disable this behavior, to ensure a fully cached list will have
-    // uncachedUnits == 0.
-    auto billAtLeastOne = completelyCached ? BillAtLeastOne::NO : BillAtLeastOne::YES;
-    uint32_t uncachedUnits = billingUnits(uncachedReadBytes, billAtLeastOne);
-    uint32_t cachedUnits = totalUnits - uncachedUnits;
-
-    actorMetrics.addUncachedStorageReadUnits(uncachedUnits);
-    actorMetrics.addCachedStorageReadUnits(cachedUnits);
-  } else {
-    // We bill 1 uncached read unit if there was no results from the list.
-    actorMetrics.addUncachedStorageReadUnits(1);
-  }
-
-  return js.v8Ref(map.As<v8::Value>());
-}
-
-kj::Function<jsg::Value(jsg::Lock&, ActorCacheOps::GetResultList)> getMultipleResultsToMap(
-    size_t numInputKeys) {
-  return [numInputKeys](jsg::Lock& js, ActorCacheOps::GetResultList value) mutable {
-    v8::HandleScope scope(js.v8Isolate);
-    auto context = js.v8Context();
-
+  return js.withinHandleScope([&] {
     auto map = v8::Map::New(js.v8Isolate);
-    uint32_t cachedUnits = 0;
-    uint32_t uncachedUnits = 0;
+    size_t cachedReadBytes = 0;
+    size_t uncachedReadBytes = 0;
     for (auto entry: value) {
-      auto& unitsRef = entry.status == ActorCacheOps::CacheStatus::CACHED
-                    ? cachedUnits : uncachedUnits;
-      unitsRef += billingUnits(entry.key.size() + entry.value.size());
-      jsg::check(map->Set(context, jsg::v8Str(js.v8Isolate, entry.key),
+      auto& bytesRef = entry.status == ActorCacheOps::CacheStatus::CACHED
+                    ? cachedReadBytes : uncachedReadBytes;
+      bytesRef += entry.key.size() + entry.value.size();
+      jsg::check(map->Set(js.v8Context(), jsg::v8Str(js.v8Isolate, entry.key),
```

Comment:
Looking at the implementation of `js.v8Context()`, I guess there's some marginal amount of overhead to getting the context value and setting up the local, but that having a consistent style of calling `js.v8Context()` everywhere is preferable? (or, perhaps, necessary for compatibility with later code changes?)

Otherwise, I'm not sure there's a particular disadvantage of leaving the variable setup lifted outside of the for loop, here and below.  Probably not a big deal either way, given that the data should be in cache while iterating.

Reply:
A lot of this is transitional. The goal is that most of the currently direct uses of `v8Context()` will be going away in favor of more ergonomic APIs and abstractions. It's just going to take a while to get there. I'm updating these now as I go to make finding and changing them later easier by ensuring the searching patterns are more consistent.